app/(candidate)/dashboard/shared/P2PUpgradeModal.js

```diff
@@ -0,0 +1,144 @@
+    imageAlt: 'Text messaging campaign illustration',
+    items: [
+      <>Send 5,000 free texts to voters</>,
+      <>Target the right voters</>,
+      <>Craft your message and image</>,
+    ],
+    highlight: null,
+    cta: 'Complete Registration',
+    href: '/profile#texting-compliance',
+  } : {
+    title: 'Level the playing field for less',
+    image: '/images/features.png',
+    imageAlt: 'Campaign features illustration',
+    items: [
+      <>Send 5,000 free texts to voters</>,
+      <>Target the right voters</>,
+      <>Craft your message and image</>,
+    ],
+    highlight: 'Join today for just $10/month.',
+    cta: 'Upgrade now',
+    href: '/dashboard/upgrade-to-pro',
+  }
+  
+  useEffect(() => {
+    if (open) {
+      trackEvent('P2P Upgrade - Modal: Modal Shown', {
+        variant,
+        modalType: 'P2PUpgrade',
+        ...trackingAttrs
+      })
+    }
+  }, [open, variant, trackingAttrs])
+
+  const handleClose = () => {
+    trackEvent('P2P Upgrade - Modal: Exit', {
+      variant,
+      modalType: 'P2PUpgrade',
+      ...trackingAttrs
+    })
+    onClose()
+  }
+
+  const handleButtonClick = () => {
+    trackEvent('P2P Upgrade - Modal: Click Button', {
+      variant,
+      modalType: 'P2PUpgrade',
+      buttonText: content.cta,
+      ...trackingAttrs
+    })
+    if (onUpgradeLinkClick) {
+      onUpgradeLinkClick()
+    } else {
+      onClose()
+    }
+  }
+
+  return (
+    <Modal
+      open={open}
+      closeCallback={handleClose}
+      preventBackdropClose
+      preventEscClose
+    >
+      <div className="p-4 sm:p-6 max-w-4xl">
+        <div className="space-y-4">
+          <div className="grid grid-cols-1 lg:grid-cols-2 gap-6 lg:gap-8 items-start">
+            {/* Content Section */}
+            <div className="order-1 lg:order-1 space-y-3">
+              <H1 className="m-0 text-2xl lg:text-3xl">{content.title}</H1>
+              
+              <Body2 className="leading-6">
+                <ul className="list-none list-inside p-0 font-sfpro m-0 space-y-1">
+                  {content.items.map((item, index) => (
+                    <li key={index} className="flex items-start">
+                      <span className="mr-2 mt-1 flex-shrink-0">✓</span>
+                      <span>{item}</span>
+                    </li>
+                  ))}
+                </ul>
+              </Body2>
+              
+              {content.highlight && (
+                <Body2 className="text-blue !font-bold mt-2">{content.highlight}</Body2>
+              )}
+            </div>
+
+            <div className="order-2 lg:order-2 relative">
+              <div className="relative w-full aspect-[3/2] lg:aspect-[4/3] rounded-lg overflow-hidden">
+                <Image
+                  src={content.image}
+                  alt={content.imageAlt}
+                  fill
+                  className={content.image.includes('features.png') ? "object-contain" : "object-cover"}
```

Comment:
basing the style on the _file name_ of the image being rendered is more than likely going to lead to a display in the bug.  Basing style off of image _features_ (ie. dimensions, mime type, etc.) would be _far_ less brittle.